smart_card_connector_app/src/pcsc-api-jstocxxtest.js

```diff
@@ -983,6 +983,125 @@ goog.exportSymbol('testPcscApi', {
       assertEquals(result.getErrorCode(), API.SCARD_S_SUCCESS);
     },
 
+    // Test `SCardConnect()` successfully connects via the "T1" protocol if the
+    // previous connection via the "RAW" protocol was terminated by
+    // `SCardDisconnect` with `SCARD_RESET_CARD`.
+    'testSCardConnect_successProtocolChangeAfterReset': async function() {
+      await launchPcscServer(
+          /*initialDevices=*/[{
+            'id': 123,
+            'type': SimulationConstants.GEMALTO_DEVICE_TYPE,
+            'cardType': SimulationConstants.COSMO_CARD_TYPE
+          }]);
+      const context = await establishContextOrThrow();
+      // Connect via the "RAW" protocol and disconnect.
+      const firstResult = await client.api.SCardConnect(
+          context, SimulationConstants.GEMALTO_PC_TWIN_READER_PCSC_NAME0,
+          API.SCARD_SHARE_SHARED,
+          /*preferred_protocols=*/ API.SCARD_PROTOCOL_RAW);
+      let sCardHandle;
+      firstResult.get(
+          (sCardHandleArg) => {
+            sCardHandle = sCardHandleArg;
+          },
+          (errorCode) => {
+            fail(`Unexpected error in first SCardConnect: ${errorCode}`);
+          });
+      const disconnectResult =
+          await client.api.SCardDisconnect(sCardHandle, API.SCARD_RESET_CARD);
+      assertEquals(disconnectResult.getErrorCode(), API.SCARD_S_SUCCESS);
+
+      // Attempt connecting via a different protocol ("ANY" denotes "either T0
+      // or T1").
+      const secondResult = await client.api.SCardConnect(
+          context, SimulationConstants.GEMALTO_PC_TWIN_READER_PCSC_NAME0,
+          API.SCARD_SHARE_SHARED,
+          /*preferred_protocols=*/ API.SCARD_PROTOCOL_ANY);
+
+      let called = false;
+      secondResult.get(
+          (sCardHandleArg, activeProtocol) => {
+            called = true;
+            assert(Number.isInteger(sCardHandle));
+            assertEquals(activeProtocol, API.SCARD_PROTOCOL_T1);
+          },
+          (errorCode) => {
+            fail(`Unexpected error in first SCardConnect: ${errorCode}`);
+          });
+      assertTrue(called);
+      assertEquals(secondResult.getErrorCode(), API.SCARD_E_PROTO_MISMATCH);
```

Comment:
why should it return error code? it looks like you expect success here.

Reply:
done, thanks